portal-api/src/v1/routes.js

```diff
@@ -44,32 +47,43 @@ openRouter.route('/users/reset-password/:resetPwdToken').post(users.resetPasswor
 
 // API Key Routes
 openRouter.route('/feedback').post(checkApiKey, feedback.create);
-openRouter.route('/user').post(checkApiKey, users.create); // This endpoint is only used by mock-data-loader, for setting up an initial user
+// This endpoint is only used by mock-data-loader, for setting up an initial user
+openRouter.route('/user').post(checkApiKey, users.create);
+
+/**
+ * Mandatory Criteria routes
+ * Allow POST & PUT of MC HTML tags
+ * on non-production environments only
+ */
+authRouter
```

Comment:
I think it would be better to have this and the routes on line 62 in a separate router. Because currently some routes on `authRouter` are subject to different checks than others which could be confusing to future developers.

We could call this router something like `nonProductionRouter` to make it clear that the endpoints on it should only be available to non-production environments. And maybe a comment to explain that they shouldn't be available on production because they are not subject to the xss check.

Reply:
Actually, if we do go for this option, we could potentially move the environment check up into `createApp` and only use the router for non-production environments. 
Then it wouldn't be possible for any endpoints using it to accidentally forget to add an environment check.

Reply:
Not all the routes are used for non production environment.

Reply:
The two routes I'm talking about are only used in non-production though right? Just the POST and the PUT?

I mean something like this:
```
const nonProductionOnlyRouter = express.Router();

nonProductionOnlyRouter
  .route('/mandatory-criteria')
  .post(passport.authenticate('jwt', { session: false }), validate({ role: ['editor'] }), mandatoryCriteria.create);

nonProductionOnlyRouter
  .route('/mandatory-criteria/:version')
  .put(passport.authenticate('jwt', { session: false }), validate({ role: ['editor'] }), mandatoryCriteria.update);

const authRouter = express.Router();
authRouter.use(passport.authenticate('jwt', { session: false }));
authRouter.use(cleanXss);

authRouter
  .route('/mandatory-criteria')
  .get(mandatoryCriteria.findAll);

authRouter
  .route('/mandatory-criteria/latest')
  .get(mandatoryCriteria.findLatest);

...
```

And then in `createApp.js` you'd have something like this:
```
app.use('/v1', openRouter);
if (process.env.NODE_ENV !== 'production') {
  app.use('/v1', nonProductionOnlyRouter)
}
app.use('/v1', authRouter);
```
So that the check for what environment we're on is brought right to the top in createApp rather than being needed to be implemented on each individual route.


But I'll leave it up to you to decide whichever way you prefer. 😄

Reply:
Thanks for the comment ill cover this in a separate PR where I'll clean the code and introduce functions ascertaining environment type. Thanks